demo/src/Root.tsx

```diff
@@ -6,16 +6,19 @@
  *
  */
 
-import { PureComponent, useState } from 'react';
-import { Button, SafeAreaView, Text, View } from 'react-native';
+import { PureComponent } from 'react';
+import { Button, SafeAreaView, View } from 'react-native';
 import NavigatorLegacy from './NavigatorLegacy';
 import Navigator from './Navigator';
 
+type Props = {};
+type State = { useLegacy: boolean };
+
 /**
  * Branch the demo/example to use new navigation or old navigation
  */
-export default class Root extends PureComponent {
-  constructor(props) {
+export default class Root extends PureComponent<Props, State> {
+  constructor(props: Props | Readonly<Props>) {
```

Comment:
I think props should always be read-only. Is this how React recommends typing props in TypeScript?